sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/DoFnSignaturesTest.java

```diff
@@ -401,6 +504,95 @@ public void foo(ProcessContext context) {}
   }
 
   @Test
+  public void testUsageOfStateDeclaredInSuperclass() throws Exception {
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("process");
+    thrown.expectMessage("declared in a different class");
```

Comment:
Should we also

`expectMessage(DoFnDeclaringStaet.class.getSimpleName())`
and
`expectMessage(the anonymous class)`?